qiskit/transpiler/target.py

```diff
@@ -322,6 +326,7 @@ def __init__(
                         "length of the input qubit_properties list"
                     )
         self.qubit_properties = qubit_properties
+        self.meas_map = format_meas_map(meas_map) if isinstance(meas_map, list) else meas_map
```

Comment:
Despite documenting the list option in the doc string, we actually convert to the dictionary option here.

Reply:
Do we need to convert the data format here? Probably this is bit confusing to users if input data is implicitly modified. Since formatted meas map is useful for implementing pulse scheduling, probably we can do this format in the scheduler. Then we can restrict the input to list format.

Reply:
@nkanazawa1989 
It was my understanding that treating meas_map as dict is easier to use for developing and it should be better to use as dict from now on. It is fine to use meas_map as list, so I will use it as list.

Reply:
I fixed at https://github.com/Qiskit/qiskit-terra/pull/10258/commits/dc454054121703310ec4abbf0f969130f5d6d2e3.